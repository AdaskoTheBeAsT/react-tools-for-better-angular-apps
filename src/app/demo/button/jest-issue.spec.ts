// @ISSUE:
// https://github.com/thymikee/jest-preset-angular/issues/128

import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { ComponentFixture, async, TestBed } from '@angular/core/testing';

@Component({
  // tslint:disable-next-line:component-selector
  selector: 'test-wrapper',
  template: `<pizza-item [name]="name" [toppings]="toppings"></pizza-item>`,
})
class TestWrapperComponent {
  name: string;
  toppings: string[];
}

@Component({
  // tslint:disable-next-line:component-selector
  selector: 'pizza-item',
  template: `
     <h3>{{name}}</h3>
    <ul><li *ngFor="let toppingName of toppings">{{toppingName}}</li></ul>
  `,
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class PizzaItemComponent {
  @Input() name = '';
  @Input() toppings: string[] = [];
}

describe('PizzaItemComponent', () => {
  let component: PizzaItemComponent;
  let fixture: ComponentFixture<PizzaItemComponent>;

  beforeEach(
    async(() => {
      TestBed.configureTestingModule({ declarations: [PizzaItemComponent] })
        .overrideComponent(PizzaItemComponent, {
          set: { changeDetection: ChangeDetectionStrategy.Default },
        })
        .compileComponents();
    })
  );

  beforeEach(() => {
    fixture = TestBed.createComponent(PizzaItemComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it(`should snapshot pizza name @Input`, () => {
    component.name = 'Chilli pizza';

    expect(fixture).toMatchSnapshot();

    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it(`should reflect toppings @Input() via *ngFor`, () => {
    component.toppings = ['first'];
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();

    component.toppings = ['anchovy', 'tomato', 'chili'];
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
