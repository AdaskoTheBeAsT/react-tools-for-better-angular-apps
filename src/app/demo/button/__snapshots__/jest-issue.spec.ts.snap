// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PizzaItemComponent should reflect toppings @Input() via *ngFor 1`] = `
<pizza-item
  name=""
  toppings={[Function Array]}
>
  
       <h3>
    
  </h3>
      <ul>
    
    <li>
      first
    </li>
  </ul>
    
</pizza-item>
`;

exports[`PizzaItemComponent should reflect toppings @Input() via *ngFor 2`] = `
<pizza-item
  name=""
  toppings={[Function Array]}
>
  
       <h3>
    
  </h3>
      <ul>
    
    <li>
      anchovy
    </li>
    <li>
      tomato
    </li>
    <li>
      chili
    </li>
  </ul>
    
</pizza-item>
`;

exports[`PizzaItemComponent should snapshot pizza name @Input 1`] = `
<pizza-item
  name={[Function String]}
  toppings={[Function Array]}
>
  
       <h3>
    
  </h3>
      <ul>
    
  </ul>
    
</pizza-item>
`;

exports[`PizzaItemComponent should snapshot pizza name @Input 2`] = `
<pizza-item
  name={[Function String]}
  toppings={[Function Array]}
>
  
       <h3>
    Chilli pizza
  </h3>
      <ul>
    
  </ul>
    
</pizza-item>
`;
